src/main/java/com/puppycrawl/tools/checkstyle/checks/whitespace/EmptyLineSeparatorCheck.java

```diff
@@ -623,6 +650,20 @@ private boolean hasEmptyLineAfter(DetailAST token) {
         return hasEmptyLine(currentEnd + 1, nextBegin - 1);
     }
 
+    /**
+     * Finds comment in next sibling of given packageDef.
+     *
+     * @param packageDef token to check
+     * @return comment under the token
+     */
+    private static Optional<DetailAST> findCommentUnder(DetailAST packageDef) {
+        return Optional.ofNullable(packageDef.getNextSibling())
+            .map(sibling -> sibling.findFirstToken(TokenTypes.MODIFIERS))
+            .map(DetailAST::getFirstChild)
+            .filter(EmptyLineSeparatorCheck::isComment)
+            .filter(comment -> comment.getLineNo() == packageDef.getLineNo() + 1);
```

Comment:
What if comment is on same line of package definition?
Example:
`package test; // example`
or 
`/* example */ package test;`

Reply:
Both cases are not reported as violation.
As I see from test cases and implementation, trailing comments are skipped

Reply:
Does it make sense to add such cases to Inputs ?
Do we have note in documentation about training or same line comments ?

Reply:
@romani 
1. This check is to make sure tokens are no in adjacent lines. If they are on same line - it seems ok in scope of this check.

2. About trailing comments - I saw nothing in docs, and may be it is ok (since trailing comments are on same line)
I saw in implementation code like 
```
DetailAST nextToken = ast.getNextSibling();
        while (nextToken != null && isComment(nextToken)) {
            nextToken = nextToken.getNextSibling();
        }
```
So the comments are skipped, even if they are not trailing

Reply:
@strkkk , sorry that so much problems sticking to simple issue.
Please do me a favor, please add note to documentation that comments are skipped, to avoid such discussions in future.

it is last item to improve.

Reply:
@strkkk , please do this last item.

Reply:
@romani I applied your changes and added note to ATTENTION paragraph.

Please take a look